addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/GreenModeHandler.java

```diff
@@ -0,0 +1,173 @@
+        return integrationId;
+    }
+
+    @Override
+    public void initialize() {
+        Number id = (Number) getThing().getConfiguration().get(INTEGRATION_ID);
+        if (id == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No integrationId");
+            return;
+        }
+        integrationId = id.intValue();
+
+        Number pollInterval = (Number) getThing().getConfiguration().get(POLL_INTERVAL);
+        if (pollInterval == null) {
+            this.pollInterval = POLL_INTERVAL_DEFAULT;
+        } else {
+            this.pollInterval = pollInterval.intValue();
+            this.pollInterval = Math.min(this.pollInterval, POLL_INTERVAL_MAX);
+            this.pollInterval = Math.max(this.pollInterval, POLL_INTERVAL_MIN);
+        }
+        logger.debug("Initializing Green Mode handler for integration ID {} with poll interval {}", integrationId,
+                this.pollInterval);
+
+        initDeviceState();
+    }
+
+    @Override
+    protected void initDeviceState() {
+        logger.debug("Initializing device state for Green Mode subsystem {}", getIntegrationId());
+        stopPolling();
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No bridge configured");
+        } else if (bridge.getStatus() == ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "Awaiting initial response");
+            queryGreenMode(ACTION_STEP); // handleUpdate() will set thing status to online when response arrives
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    @Override
+    protected void thingOfflineNotify() {
+        stopPolling();
+    }
+
+    private void startPolling() {
+        if (pollInterval > 0 && pollJob == null) {
+            logger.debug("Scheduling green mode polling job for integration ID {}", integrationId);
+            pollJob = scheduler.scheduleWithFixedDelay(this::pollState, pollInterval, pollInterval, TimeUnit.MINUTES);
+        }
+    }
+
+    private void stopPolling() {
+        if (pollJob != null) {
+            logger.debug("Canceling green mode polling job for integration ID {}", integrationId);
+            pollJob.cancel(true);
+            pollJob = null;
+        }
+    }
+
+    private synchronized void pollState() {
+        logger.trace("Executing green mode polling job for integration ID {}", integrationId);
+        queryGreenMode(ACTION_STEP);
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        if (channelUID.getId().equals(CHANNEL_STEP)) {
+            queryGreenMode(ACTION_STEP);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(CHANNEL_STEP)) {
+            if (command.equals(OnOffType.ON)) {
```

Comment:
Its an enum you could use ==

Reply:
Yeah, since command can't be null I was indifferent about which to use.  I guess == is a bit nicer.